src/Composer/Installers/CakePHPInstaller.php

```diff
@@ -34,7 +34,7 @@ public function inflectPackageVars($vars)
     public function getLocations()
     {
         if ($this->matchesCakeVersion('>=', '3.0.0')) {
-            $this->locations['plugin'] = 'plugins/{$name}/';
+            $this->locations['plugin'] = 'vendor/{$name}/';
```

Comment:
I think this will cause the plugin to be installer under `vendor/PluginName` instead of `vendor/vendor-name/plugin-name`.